src/t_zset.c

```diff
@@ -4246,16 +4246,15 @@ void zrandmemberCommand(client *c) {
 
     if (c->argc >= 3) {
         if (getRangeLongFromObjectOrReply(c, c->argv[2], -LONG_MAX, LONG_MAX, &l, NULL) != C_OK) return;
+        if (l < -server.max_rand_count || l > server.max_rand_count) {
+            addReplyError(c, "value is out of range");
+            return;
+        }
         if (c->argc > 4 || (c->argc == 4 && strcasecmp(c->argv[3]->ptr, "withscores"))) {
             addReplyErrorObject(c, shared.syntaxerr);
             return;
-        } else if (c->argc == 4) {
+        } else if (c->argc == 4)
             withscores = 1;
```

Comment:
suggest it is clearer with the curly brackets
```suggestion
        } else if (c->argc == 4) {
            withscores = 1;
        }
```